lumenlearning/lti-middleware
# L3-68 fix issue of not allowing multiple deployments

## Description
Moved validation of issuer, clientId, and deploymentId from OIDC request controller to LTI launch controller because the LTI 1.3 Core specification does not guarantee that the clientId or deploymentId will be available for validation until after the id_token has been retrieved.

### Motivation and Context
This change will allow a tool have multiple deployments associated with a single issuer and/or clientId without receiving a validation error.

### Fixes
[L3-68](https://lumenlearning.atlassian.net/browse/L3-68)

## How Has This Been Tested?
1. Ensure there is a functional Developer Key for the lti-middleware in Canvas.
2. Using the same client id from this Developer Key, deploy the lti-middleware to two separate Canvas courses and retrieve the deploymentId from each.
3. Perform a POST on the /config endpoint of the middleware for each deployment configuration.
4. Click on an LTI link in each of the two courses and ensure that the flow occurs as expected without encountering any error pages from the middleware.

---

## Checklist
